packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -100,9 +101,11 @@ const Tooltip: React.FC<TooltipOwnProps> = (props) => {
       if (context.id === contentId) {
         if (state === 'open') {
           setStateAttribute(context.delayed ? 'delayed-open' : 'instant-open');
-        } else if (state === 'closed') {
+        } else {
```

Comment:
Removed this explicit check, that's what was breaking the unmount animation. It was there because that PR was based on having only `open` and `closed` state, but now that we have more it should go (wasn't needed anyway).